src/main/java/jimmy/mcgymmy/model/ModelManager.java

```diff
@@ -114,6 +116,23 @@ public void setFood(Index index, Food editedFood) {
         mcGymmy.setFood(index, editedFood);
     }
 
+    @Override
+    public void clearFilteredFood() {
+        List<Food> lst = new ArrayList<>();
+
```

Comment:
I think we can change this to a for loop where we delete using the `filteredList`'s predicate, i.e.

```java
var predicate = filteredFoodItems.getPredicate();
for (Food food : listOfFoodSomewhere) {
    if (predicate.check(food)) {
      removeItem(food);
    }
}
```
This will work as intended, as long as `getPredicate` and `setPredicate` in `fxcollections` works as intended

Reply:
Something along this line instead?

```
var predicate = filteredFoodItems.getPredicate();
newList = listOfFoodSomewhere.stream.filter((x) -> !predicate(food)).collect(Collectors.toList());
setMcGymmy(new McGymmy(newList));
```

Reply:
Can also create a `public void filter(Predicate p)` in `Fridge.java` then call it somehow in this method.
Then we can use java streams to filter and override the list (but don't forget to invert the boolean predicate).

Reply:
the filtered list is implemented in Model so I think it will be better to implement in model instead of in Fridge.
Can be a possible future optimization